std/asyncio/net/UdpSocket.hx

```diff
@@ -0,0 +1,93 @@
+package asyncio.net;
+
+import asyncio.net.SocketOption.SocketOptionKind;
+import haxe.io.Bytes;
+import haxe.NoData;
+import haxe.errors.NotImplemented;
+import haxe.Callback;
+
+class UdpSocket {
+	/**
+		Indicates if the socket is currently bound to a specific remote address.
+	**/
+	public var bound(get,never):Bool;
+	function get_bound():Bool throw new NotImplemented();
+
+	/**
+		Local address of this socket.
+	**/
+	public var localAddress(get,never):{host:String, port:Int};
+	function get_localAddress():{host:String, port:Int} throw new NotImplemented();
+
+	/**
+		Remote address of this socket if it is bound.
+	**/
+	public var remoteAddress(get,never):Null<{host:String, port:Int}>;
+	function get_remoteAddress():Null<{host:String, port:Int}> throw new NotImplemented();
+
+	/**
+		Open a UDP socket.
+	**/
+	static public function open(?address:{host:String, port:Int}, ?options:Array<SocketOption>, callback:Callback<Null<UdpSocket>>) {
+		callback.fail(new NotImplemented());
+	}
+
+	/**
+		Bind the socket to the `host` and `port`.
+		The callback is supplied with the new write function, which allows to send
+		data without the need to specify remote address on each call.
+	**/
+	public function bind(host:String, port:Int, callback:Callback<Null< (buffer:Bytes, offset:Int, length:Int)->Void >>) {
```

Comment:
I find this a bit strange. To bind you call this function and get back a function which is the "bound" send for this socket. But at the same time the socket object itself is bound, so further `read`s should come come from the bound port, and you can unbind the object without knowing the function. What happens if you call the function after the object is unbound?

Reply:
Indeed. 
Maybe `bind` should give an `IDuplex` instance with both `read` and `write` and also `unbound`, so that it would be impossible to call `UdpSocket.unbound` on an already unbound socket.
I'd like to avoid situations when we tell user "don't do this if in that state". Instead I want to provide a specific API for each specific state.